tenant/storage_urm.go

```diff
@@ -13,7 +14,57 @@ var (
 	urmByUserIndexBucket = []byte("userresourcemappingsbyuserindexv1")
 )
 
+func getBucket(rt influxdb.ResourceType) ([]byte, error) {
+	var bucket []byte
+	switch rt {
+	case influxdb.UsersResourceType:
+		bucket = userBucket
+	case influxdb.OrgsResourceType:
+		bucket = organizationBucket
+	case influxdb.BucketsResourceType:
+		bucket = bucketBucket
+	default:
+		return nil, &influxdb.Error{
+			Code: influxdb.EInternal,
+			Msg:  fmt.Sprintf("cannot get bucket of resource type: %v", rt),
+		}
+	}
+	return bucket, nil
+}
+
+func errNotFound(rt influxdb.ResourceType) error {
+	switch rt {
+	case influxdb.UsersResourceType:
+		return ErrUserNotFound
+	case influxdb.OrgsResourceType:
+		return ErrOrgNotFound
+	case influxdb.BucketsResourceType:
+		return ErrBucketNotFound
+	default:
+		return &influxdb.Error{
+			Code: influxdb.EInternal,
+			Msg:  fmt.Sprintf("cannot get bucket of resource type: %v", rt),
+		}
+	}
+}
+
 func (s *Store) CreateURM(ctx context.Context, tx kv.Tx, urm *influxdb.UserResourceMapping) error {
+	resBucket, err := getBucket(urm.ResourceType)
+	if err != nil {
+		return err
+	}
+	if _, err := s.get(ctx, tx, userBucket, urm.UserID); err != nil {
+		if kv.IsNotFound(err) {
+			return errNotFound(influxdb.UsersResourceType)
+		}
+		return err
+	}
+	if _, err := s.get(ctx, tx, resBucket, urm.ResourceID); err != nil {
+		if kv.IsNotFound(err) {
+			return errNotFound(urm.ResourceType)
+		}
+		return err
+	}
```

Comment:
I think its a good thing to make sure the user exist for the URM. I dont think once we have other systems (like auth/task) using create URM we will not be able to verify that the resource exists before creating a urm. I think the right action here is to check user (probably using the `s.GetUser` function instead of a generic get) and then create the urm weather the other resource exists or not. a dangling urm isn't a problem I dont think

Reply:
That's perfect! I was investing time in making most tests pass given this new logic. I'll rethink about that and update. Thank you

Reply:
Thinking more about it, this matches with the problem with the `switch` statement above.

I see one way of doing this: refactor the switch not to return a `bucket`,  but to implement a generic resource find that errors on `not found`. The problem there is that we would end up having the storage layer depend on the service layer. We don't want that. That check should be done at the service layer 👍 

For now, I'll do what you suggest. I'll check user and not the other resource 👍